packages/protocol/src/vaults/borrowing/BorrowingVault.sol

```diff
@@ -301,6 +303,37 @@ contract BorrowingVault is BaseVault {
     return shares;
   }
 
+  /**
+   * @dev Checks if vault debt has been paid off externally and pauses the vault if so.
+   * Will call withdraw if normal conditions are met.
+   *
+   * @param caller or {msg.sender}
+   * @param receiver to whom `assets` amount will be transferred to
+   * @param owner to whom `shares` will be burned
+   * @param assets amount transferred during this withraw
+   * @param shares amount burned to `owner` during this withdraw
+   */
+  function _withdraw(
+    address caller,
+    address receiver,
+    address owner,
+    uint256 assets,
+    uint256 shares
+  )
+    internal
+    override
+    whenNotPaused(VaultActions.Withdraw)
+  {
+    uint256 totalDebt_ = totalDebt();
+    uint256 supply = debtSharesSupply;
+
+    if (totalDebt_ == 0 && supply > 0 && supply > totalDebt_) {
+      _pause(VaultActions.Withdraw);
+      revert BorrowingVault__withdraw_debtNeedsCorrection();
```

Comment:
can you please explain what you intend to do here?
I see a pause on top and a revert added just below

Reply:
This is the situation in where some external actor, pays entirely the debt of the vault. By that previously, they will effectively, disable the vault from borrowing operations, if I recall. 
Now, if... someone, does pay the vault's debt ever, we created a function to restore debt back, and essentially later decide what to do with the "benefactor" proceeds. 
The conditions there, is to avoid withdraws at the moment right after the debt has been paid back.

If you read the conditions: if total debt is zero, meaning... someone paidback the whole debt, and there is debtSharesSupply, meaning.... there were user/s owing funds. Then pause all withdrawals until all gets sort out.

In the case of someone paying off all the debt, we don't want to create "bad debt" position, by allowing users to withdraw the shares and leaving "unbacked" debtShares, once debt is restored.

Reply:
Though the revert will do effectively the job. I see now, that state for pause will not persist.

Reply:
yep general intention makes sense 

> Though the revert will do effectively the job. I see now, that state for pause will not persist.

I was more concerned with the above, yep it won't persist

Reply:
Refer to #536